### ISSUE TITLE
Instance Refresh Example: The "count" value depends on resource attributes that cannot be determined

### ISSUE THREAD

**Issue by jaimehrubiks** [terraform-aws-modules/terraform-aws-eks#1393]
## Description

Add instance refresh code to my existing cluster. The "aws_autoscaling_lifecycle_hook" code piece is failing.

Related to
https://github.com/terraform-aws-modules/terraform-aws-eks/pull/1373#discussion_r635271679

## ⚠️ Note

Already tried deleting .terraform, init and plan/apply

## Versions
 - Terraform: v0.13.6
 - Provider(s):
❯ terraform providers --version
Terraform v0.13.6
```
+ provider registry.terraform.io/hashicorp/aws v3.37.0
+ provider registry.terraform.io/hashicorp/cloudinit v2.2.0
+ provider registry.terraform.io/hashicorp/helm v2.1.2
+ provider registry.terraform.io/hashicorp/kubernetes v1.11.4
+ provider registry.terraform.io/hashicorp/local v1.4.0
+ provider registry.terraform.io/hashicorp/null v2.1.2
+ provider registry.terraform.io/hashicorp/random v2.3.1
+ provider registry.terraform.io/hashicorp/template v2.2.0
+ provider registry.terraform.io/terraform-aws-modules/http v2.4.1
```
 - Module: latest (16.2.0)

## Reproduction

Add
```
resource "aws_autoscaling_lifecycle_hook" "aws_node_termination_handler" {
  count                  = length(module.eks.workers_asg_names)
  name                   = "aws-node-termination-handler"
  autoscaling_group_name = module.eks.workers_asg_names[count.index]
  lifecycle_transition   = "autoscaling:EC2_INSTANCE_TERMINATING"
  heartbeat_timeout      = 300
  default_result         = "CONTINUE"
}
```

### Terminal Output Screenshot(s)

```
Error: Invalid count argument

  on platform.tf line 25, in resource "aws_autoscaling_lifecycle_hook" "aws_node_termination_handler":
  25:   count                  = length(module.eks.workers_asg_names)

The "count" value depends on resource attributes that cannot be determined
until apply, so Terraform cannot predict how many instances will be created.
To work around this, use the -target argument to first apply only the
resources that the count depends on.
```

**Comment by jaimehrubiks** [comment]
I realized what could be the issue here.

I recently migrated from terraform 0.12 to terraform 0.13, and I think there is an issue with terraform 0.13.

    terraform plan -refresh=false

doesn't show any error, which I think they changed that behavior in 0.13

I'm starting to think that terraform 0.12 was much more consistent than 0.13. 

Terraform 0.14 seems to have fixed this issue, so I will try 0.14 today/tomorrow and comment back.
